simapp/codec.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package cosmos_sdk.simapp;
+option go_package = "simapp";
+import "gogoproto/gogo.proto";
+import "types/codec.proto";
+import "x/gov/types/codec.proto";
+
+message Msg {
```

Comment:
Is there a reason this is defined in `simapp/` and not local to where the types are defined (i.e. `x/gov/types`)?

Reply:
So this is the `Msg` `oneof` for `simapp`. Each app would have a different `Msg` struct and that would likely be used directly for routing rather than `Route` and `Type on the `Msg` interface. You would likely just need `ValidateBasic` and `GetSigners` on the `Msg` interface